module/cloud-services/files/helper.js

```diff
@@ -119,50 +143,125 @@ module.exports = class FilesHelper {
+        });
       } catch (error) {
-
         return reject({
           status:
-            error.status ||
-            httpStatusCode["internal_server_error"].status,
+            error.status || httpStatusCode["internal_server_error"].status,
 
           message:
-            error.message
-            || httpStatusCode["internal_server_error"].message,
-
-          errorObject: error
-        })
+            error.message || httpStatusCode["internal_server_error"].message,
 
+          errorObject: error,
+        });
       }
-    })
-
+    });
   }
-}
 
+  /**
+   * upload file to the Cloud .
+   * @method
+   * @name upload
+   * @param {Buffer} payloadData    - Binary value of file.
+   * @param {String} folderPath         - folderPath.
+   * @param {String} fileName           - fileName.
+   * @returns {JSON}                    -  path and downloadUrl of the file.
+   */
 
+  static upload(payloadData,folderPath,fileName) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let currentMoment = moment(new Date());
+        let formattedDate = currentMoment.format("DD-MM-YYYY");
+        //Create public Directory if it doesn't exist
+          this.createDirectoryForUploadFile();
+        if (!fs.existsSync(`${ROOT_PATH}/public/assets/${formattedDate}`)) {
+          fs.mkdirSync(`${ROOT_PATH}/public/assets/${formattedDate}`);
+        }
+        let localFilePath = `${ROOT_PATH}/public/assets/${formattedDate}/${fileName}`;
+
+        // Use the Promise version of fs.writeFile
+        await fs.promises.writeFile(localFilePath, payloadData);
+
+        // Use fs.promises.readFile to read the file content asynchronously
+        let binaryDataOfFile = await fs.promises.readFile(localFilePath);
+
+
+        let uploadFile = await filesHelpers.upload(
+          folderPath,
+          bucketName,
+          binaryDataOfFile
+        );
+          
+       // Use fs.promises.unlink to remove the file asynchronously
+        await fs.promises.unlink(localFilePath);
+
+        if (!uploadFile.success) {
+          return resolve({
+            status: httpStatusCode["bad_request"].status,
+            message: constants.apiResponses.FAILED_TO_UPLOAD,
+            result: {},
+          });
+        }
 
+        return resolve({
+          message: constants.apiResponses.CLOUD_SERVICE_SUCCESS_MESSAGE,
+          result: uploadFile.result,
+        });
+      } catch (error) {
+        return reject({
+          status:
+            error.status || httpStatusCode["internal_server_error"].status,
+
+          message:
+            error.message || httpStatusCode["internal_server_error"].message,
+
+          errorObject: error,
+        });
+      }
+    });
+  }
+
+  /**
+   * create public/asstes folder if its doesn't exits.
+   * @method
+   * @name createDirectoryForUploadFile
+   */
+
+  static createDirectoryForUploadFile(){
+   
+  if (!fs.existsSync(ROOT_PATH + "/public")) {
```

Comment:
why is this added here it should be global js file @praveenKDass 
as if condition only

Reply:
Moved to global.js file